example/spec/parallel/moveGranuleCollections/ChangeGranuleCollectionS3.js

```diff
@@ -0,0 +1,181 @@
+  let testDataFolder;
+  let granuleId;
+  let startingFiles;
+  let finalFiles;
+  let collection;
+  let targetCollection;
+  let ingestExecutionArn;
+  let cleanupCollectionId;
+  afterAll(async () => {
+    try {
+      await removePublishedGranule({
+        prefix: config.stackName,
+        granuleId,
+        collectionId: cleanupCollectionId,
+      });
+      let cleanup = finalFiles.map((fileObj) => deleteS3Object(
+        fileObj.bucket,
+        fileObj.key
+      ));
+      cleanup.concat(startingFiles.map((fileObj) => deleteS3Object(
+        fileObj.bucket,
+        fileObj.key
+      )));
+      cleanup = cleanup.concat([
+        deleteExecution({ prefix: config.stackName, executionArn: ingestExecutionArn }),
+        deleteGranule({ prefix: config.stackName, granuleId: granuleId }),
+      ]);
+
+      await Promise.all(cleanup);
+    } catch (error) {
+      console.log('cleanup failed with error', error);
+    }
+  });
+  beforeAll(async () => {
+    const inputPayloadFilename = './spec/parallel/ingestGranule/IngestGranule.input.payload.json';
+    const providersDir = './data/providers/s3/';
+    const s3data = [
+      '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
+      '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf',
+      '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606_ndvi.jpg',
+    ];
+
+    const collectionsDir = './data/collections/s3_MOD09GQ_006_full_ingest';
+    const targetCollectionsDir = './data/collections/s3_MOD09GQ_007_full_ingest_move';
+    const granuleRegex = '^MOD09GQ\\.A[\\d]{7}\\.[\\w]{6}\\.006\\.[\\d]{13}$';
+    config = await loadConfig();
+    stackName = config.stackName;
+    const testId = createTimestampedTestId(stackName, 'IngestGranuleSuccess');
+    const testSuffix = createTestSuffix(testId);
+    testDataFolder = createTestDataPath(testId);
+
+    collection = { name: `MOD09GQ${testSuffix}`, version: '006' };
+    targetCollection = { name: `MOD09GQ${testSuffix}`, version: '007' };
+    provider = { id: `s3_provider${testSuffix}` };
+
+    // populate collections, providers and test data
+    await Promise.all([
+      uploadTestDataToBucket(config.bucket, s3data, testDataFolder),
+      addCollections(stackName, config.bucket, collectionsDir, testSuffix, testId),
+      addCollections(stackName, config.bucket, targetCollectionsDir, testSuffix, testId),
+      addProviders(stackName, config.bucket, providersDir, config.bucket, testSuffix),
+    ]);
+
+    const inputPayloadJson = fs.readFileSync(inputPayloadFilename, 'utf8');
+    // update test data filepaths
+    inputPayload = await setupTestGranuleForIngest(
+      config.bucket,
+      JSON.stringify({ ...JSON.parse(inputPayloadJson), pdr: undefined }),
+      granuleRegex,
+      testSuffix,
+      testDataFolder
+    );
+    granuleId = inputPayload.granules[0].granuleId;
+
+    ingestExecutionArn = await buildAndStartWorkflow(
```

Comment:
I think we could probably make this go faster and/or reduce the potential for random failures if we just setup the granule directly/via API rather than run a workflow, particularly given this is a single Lambda invocation.       I'm going to take a stab at that.

Reply:
@etcart  PR up https://github.com/nasa/cumulus/pull/3918